src/components/ProtectedRoute.tsx

```diff
@@ -5,11 +5,13 @@ import { Container, Typography, CircularProgress } from '@mui/material';
 import { useTranslation } from 'react-i18next';
 import useAuth from '../lib/useAuth';
 
-function ProtectedRoute({ element }) {
+function ProtectedRoute({ element }: { element: JSX.Element }): JSX.Element {
   const location = useLocation();
-  const { showLoginDialog } = location.state || {
-    showLoginDialog: undefined,
-  };
+  const { showLoginDialog }: any =
+    location.state ||
+    ({
+      showLoginDialog: undefined,
+    } as unknown as boolean | undefined);
```

Comment:
Does this work?

```suggestion
  const showLoginDialog = location.state?.showLoginDialog;
```
Might need to do `(state as any)?`